app/Http/Controllers/HomeController.php

```diff
@@ -6,39 +6,59 @@
 use App\Models\Course;
 use App\Models\User;
 use App\Models\Advisor;
+use App\Models\Message;
 use Auth;
+use App\Events\MessageEvent;
+use App\Events\MyEvent;
+use App\Repositories\Course\CourseRepositoryInterface;
+use App\Repositories\User\UserRepositoryInterface;
+use App\Repositories\Advisor\AdvisorRepositoryInterface;
+use App\Repositories\Message\MessageRepositoryInterface;
 
 class HomeController extends Controller
 {
+    protected $courseRepo, $userRepo, $advisorRepo, $messageRepo;
+
+    public function __construct(
+        CourseRepositoryInterface $courseRepo,
+        UserRepositoryInterface $userRepo,
+        AdvisorRepositoryInterface $advisorRepo,
+        MessageRepositoryInterface $messageRepo
+    ) {
+        $this->courseRepo = $courseRepo;
+        $this->userRepo = $userRepo;
+        $this->advisorRepo = $advisorRepo;
+        $this->messageRepo = $messageRepo;
+    }
     /**
      * Show the application dashboard.
      *
      * @return \Illuminate\Contracts\Support\Renderable
      */
     public function index()
     {
-        $courses = Course::with('image')->latest('created_at')->take(config('title.course_number'))->get();
-        
+        $courses = $this->courseRepo->getLatestCourse(['image'], config('title.course_number'));
+
         return view('user.component.index', compact('courses'));
     }
 
     public function course()
     {
-        $courses = Course::with('image')->get();
+        $courses = $this->courseRepo->getAll(['image']);
         
         return view('user.component.course', compact('courses'));
     }
 
-    public function showLessons(Course $course)
+    public function showLessons($course)
     {
-        $course->load(['image', 'lessons', 'comments.user.image']);
+        $course = $this->courseRepo->loadRelations($course, ['image', 'lessons', 'comments.user.image']);
         
         return view('user.component.lessons', compact('course'));
     }
 
     public function showMentors()
     {   
-        $mentors = User::with('mentorComments.user.image', 'image')->where('role_id', config('role.mentor_id'))->get();
+        $mentors = $this->userRepo->getAllWithWhere(['mentorComments.user.image', 'image'], ['role_id' => config('role.mentor_id')]);
```

Comment:
```suggestion
        $mentors = $this->userRepo->getData(['mentorComments.user.image', 'image'], ['role_id' => config('role.mentor_id')]);
```
sửa lại tên function nhé